### PR TITLE
support for opentsdb /api/query/exp

### LINKED ISSUE #3805: Feature Request: Support for OpenTSDB 2.3 Expressions
The company I worked for submitted the expression code and has an internal Datasource that works with the expressions, although, it's simplistic at best. It is simply a text box where you write your expression to be returned.

I'm planning to submit a PR to add that datasource, but if @utkarshcmu has something better in the works, like improvements to the OpenTSDB datasource to support them the way the Graphite or Bosun datasources do, that would be great.

An example of the new expression usage:

``` bash
curl 'http://localhost:4244/api/query/gexp/?start=10m-ago&exp=alias(scale(sum:rate:1m-avg:tsd.rpc.received\{domain=atl1\},0.10),TSDBRPCSCALED)' -H 'Accept: application/json, text/plain, */*' -H 'Referer: http://localhost/grafana/' -H 'Origin: http://localhost' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36' --compressed
```

A picture of our Datasource in Use:

grafana-opentsdb-expression

Originally I posted this in #3077 but I realized that was for the filters not the new expressions.

### LINKED ISSUE #6348: Add Edit mode for OpenTSDB queries
- **I'm submitting a ...**
- [ ] Bug report
- [x] Feature request
- [ ] Question / Support request: **Please do not** open a github issue. [Support Options](http://grafana.org/support/)

We should have edit mode for OpenTSDB queries as well similar to graphite. Once done, it will be a good base for supporting expressions. #3805

### LINKED ISSUE #7754: OpenTSDB: Support for OpenTSDB & Multi-value metric selection
The metrics command works as expected without checking the multi-value option.
After enabling the multi-value option I get the following error:

"error": {
 "code": 400,
 "message": "No such name for 'metrics': '{firstmetric,secondmetric}'",
...

Please include this information:

* What Grafana version are you using? -- v4.1.1
* What datasource are you using? --- OpenTSDB v2.2 (Zenoss)

Thanks for any help/reply to further investigate this behaviour.

### PR DESCRIPTION
**Pull request by gnydick**
Fixes #3805
Fixes #6348
Fixes #7754

I’ve got a pretty good bit of functionality patched in adding support in opentsdb for /api/query/exp. I think the opentsdb users would LOVE this capability, but it’s 99.9% of the way there, I don’t have any interest in writing unit tests as I’m not even a TS developer, but I got through it, but I think this contrib should be nurtured through by someone to make it 100% clean enough for mainlining. 

The only thing I can think of that's missing is that the layout for the form to enter the expression needs to be fixed. Otherwise, it's a treat ;)

FWIW, I believe this should merge relatively cleanly into 6.0 as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -54,7 +54,6 @@ profile.cov
 /pkg/cmd/grafana-cli/grafana-cli
 /pkg/cmd/grafana-server/grafana-server
 /pkg/cmd/grafana-server/debug
-/pkg/extensions
```
Comment (marefr) [inline .gitignore]: Please revert this change. It's needed even though it mat not look like that.

**Review by marefr — CHANGES_REQUESTED** [review]
Did a quick look at this. Added a comment. 

Besides that, I see three problems with proposed PR:
1. You should target this PR to be merged to master. I've changed that. Please merge your branch with master
2. Don't understand why you got updates in pkg/vendor (maybe because you're targeting v5.4.x branch instead of master)
3. We would require you to do required changes in backend (pkg/tsdb/opentsdb.go) to support using opentsdb /api/query/exp with Grafana alerting

**Comment by gnydick** [comment]
Sounds good, I should be able to make those changes.

File: .gitignore (hunk 1 above)
Comment (gnydick) [reply .gitignore]: sure thing, but when I was working, that directory kept going missing. not sure why, and so my local dev stopped working.

**Comment by gnydick** [comment]
@marefr I'm not following, i see that you 'self-assigned' stuff. Does that mean you're taking care of all of the changes requested here, and I just need to do the backend changes?

**Comment by marefr** [comment]
@gnydick I'm assigned to this PR as responsible for reviewing and helping you if you have any questions.

**Comment by NikolaBorisov** [comment]
Any chance to get this merged?

**Comment by gnydick** [comment]
I don't have the time to polish it all out. Is there a process for
requesting a new owner?
